src/gui/AutomatableModelView.cpp

```diff
@@ -258,7 +279,227 @@ void AutomatableModelViewSlots::removeConnection()
+	// a nullptr if it can not find and add a track
+	if (track == nullptr) { return; }
+
+	// getting nearest node position
+	AutomationClip* nodeClip = nullptr;
+	TimePos nodePos = getNearestAutomationNode(track, &nodeClip);
+	if (nodeClip != nullptr)
+	{
+		// modifying its value
+		nodeClip->recordValue(nodePos, m_amv->modelUntyped()->getTrueValue());
+	}
+}
+void AutomatableModelViewSlots::removeSongNearestAutomationNode()
+{
+	std::vector<AutomationClip*> clips = AutomationClip::clipsForModel(m_amv->modelUntyped());
+	// getting the track without adding a new one if no track was found
+	AutomationTrack* track = getCurrentAutomationTrack(&clips, false);
+
+	// this needs to be checked because getCurrentAutomationTrack might give
+	// a nullptr if it can not find and add a track
+	if (track == nullptr) { return; }
+
+	AutomationClip* nodeClip = nullptr;
+	TimePos nodePos = getNearestAutomationNode(track, &nodeClip);
+	if (nodeClip != nullptr)
+	{
+		nodeClip->removeNode(nodePos);
+		// if there is no node left, the automationClip will be deleted
+		if (nodeClip->hasAutomation() == false)
+		{
+			delete nodeClip;
+		}
+	}
+}
+AutomationTrack* AutomatableModelViewSlots::getCurrentAutomationTrack(std::vector<AutomationClip*>* clips, bool canAddNewTrack)
+{
+	AutomationTrack* output = nullptr;
+	if (clips->size() > 0)
+	{
+		// selecting the track with the most amount of clips
+		// connected to this model
+		AutomationTrack* maxTrack = dynamic_cast<AutomationTrack*>((*clips)[0]->getTrack());
+		int maxTrackCount = 1;
+		for (size_t i = 1; i < clips->size(); i++)
+		{
+			int currentCount = 0;
+			for (size_t j = 0; j < clips->size(); j++)
+			{
+				if ((*clips)[i]->getTrack() == (*clips)[j]->getTrack())
+				{
+					currentCount++;
+				}
+			}
+			if (maxTrackCount < currentCount)
+			{
+				maxTrackCount = currentCount;
+				maxTrack = dynamic_cast<AutomationTrack*>((*clips)[i]->getTrack());;
+			}
+		}
+		output = maxTrack;
+	}
+	else if (canAddNewTrack == true)
+	{
+		// adding new track
+		output = new AutomationTrack(getGUI()->songEditor()->m_editor->model(), false);
+	}
+	return output;
+}
+AutomationClip* AutomatableModelViewSlots::getCurrentAutomationClip(AutomationTrack* track, bool canAddNewClip, bool searchAfter)
+{
+	AutomationClip* output = nullptr;
+	std::vector<Clip*> trackClips = track->getClips();
+	TimePos timePos = static_cast<TimePos>(Engine::getSong()->getPlayPos());
+	
 
+	bool tryAdding = false;
+	if (trackClips.size() > 0)
+	{
+		// getting the closest clip that start before or after the global time position
+		tick_t closestTime = -1;
+		Clip* closestClip = nullptr;
+		for (Clip* currentClip : trackClips)
+		{
+			tick_t currentTime = currentClip->startPosition().getTicks();
+			if ((searchAfter == false && (currentTime > closestTime || closestTime < 0) && timePos.getTicks() > currentTime)
+				|| (searchAfter == true && (currentTime < closestTime || closestTime < 0) && timePos.getTicks() <= currentTime))
```

Comment:
I hate how many checks there are in this if. Please decompose them to something like this.
```suggestion
			auto currentTick = timePos.getTicks();
			bool negativeTime = closestTime < 0;
			bool beforeTime = (negativeTime || currentTime > closestTime) && currentTick > currentTime;
			bool afterTime = (negativeTime || currentTime < closestTime) && currentTick <= currentTime;
			bool beforeCheck = searchAfter == false && beforeTime;
			bool afterCheck = searchAfter && afterTime;
			if (beforeCheck || afterCheck)
```